configure.ac

```diff
@@ -251,28 +251,38 @@ AC_TYPE_INT16_T
 AC_TYPE_INT32_T
 AC_TYPE_INT64_T
 
+PKG_CHECK_MODULES([libqb], [libqb])
+
 if test "x$enable_man" = "xyes"; then
 	AC_ARG_VAR([DOXYGEN], [override doxygen executable])
 	AC_CHECK_PROGS([DOXYGEN], [doxygen], [no])
 	if test "x$DOXYGEN" = xno; then
 		AC_MSG_ERROR(["Doxygen command not found"])
 	fi
-	# required by doxyxml to build man pages dynamically
-	# Don't let AC_PROC_CC (invoked by AX_PROG_CC_FOR_BUILD) replace
-	# undefined CFLAGS_FOR_BUILD with -g -O2, overriding our special OPT_CFLAGS.
-	: ${CFLAGS_FOR_BUILD=""}
-	AX_PROG_CC_FOR_BUILD
-	saved_PKG_CONFIG="$PKG_CONFIG"
-	saved_ac_cv_path_PKG_CONFIG="$ac_cv_path_PKG_CONFIG"
-	unset PKG_CONFIG ac_cv_path_PKG_CONFIG
-	AC_PATH_PROG([PKG_CONFIG], [pkg-config])
-	PKG_CHECK_MODULES([libqb_BUILD], [libqb])
-	PKG_CHECK_MODULES([libxml_BUILD], [libxml-2.0])
-	PKG_CONFIG="$saved_PKG_CONFIG"
-	ac_cv_path_PKG_CONFIG="$saved_ac_cv_path_PKG_CONFIG"
-fi
 
-PKG_CHECK_MODULES([libqb], [libqb])
+	AC_ARG_VAR([DOXYGEN2MAN], [override doxygen2man executable])
+
+	# required to detect doxygen2man when libqb is installed
+	# in non standard paths
+	if test "x$cross_compiling" = "xno"; then
+		PKG_CHECK_VAR([libqb_PREFIX], [libqb], [prefix])
+		AC_PATH_PROG([DOXYGEN2MAN], [doxygen2man], [no], [$libqb_PREFIX/bin$PATH_SEPARATOR$PATH])
+	else
+		saved_PKG_CONFIG="$PKG_CONFIG"
+		saved_ac_cv_path_PKG_CONFIG="$ac_cv_path_PKG_CONFIG"
+		unset PKG_CONFIG ac_cv_path_PKG_CONFIG
+		AC_PATH_PROG([PKG_CONFIG], [pkg-config])
+		PKG_CHECK_MODULES([libqb_BUILD], [libqb])
+		PKG_CHECK_VAR([libqb_BUILD_PREFIX], [libqb], [prefix])
+		AC_PATH_PROG([DOXYGEN2MAN], [doxygen2man], [no], [$libqb_BUILD_PREFIX/bin$PATH_SEPARATOR$PATH])
+		PKG_CONFIG="$saved_PKG_CONFIG"
+		ac_cv_path_PKG_CONFIG="$saved_ac_cv_path_PKG_CONFIG"
```

Comment:
Isn't this code path perfectly valid even when not cross-compiling? If so, it would be advantageous to have a single path to avoid code rot.

Reply:
good point..

Reply:
done